python/chemiscope/explore.py

```diff
@@ -163,3 +173,194 @@ def soap_pca_featurize(frames):
     # Compute pca
     pca = PCA(n_components=2)
     return pca.fit_transform(feats)
+
+
+def torchscript_featurizer(model, extensions_directory="extensions/", device=None):
+    """
+    Create a featurizer function using a `metatrain`_  model.
+
+    :param model: model to use for the calculation. It can be a file path, a Python
+        instance of `MetatensorAtomisticModel
+        <https://docs.metatensor.org/latest/atomistic/reference/models/export.html#metatensor.torch.atomistic.MetatensorAtomisticModel>`_,
+        or the output of `torch.jit.script`_ on
+        :py:class:`MetatensorAtomisticModel`.
+
+    :param extensions_directory: if the model uses extensions, we will try to load
+        them from this directory, default `"extensions/"`.
+
+    :param device: torch device to use for the calculation. If ``None``, we will try
+        the options in the model's ``supported_device`` in order.
+
+    :returns featurize: a function that takes a list of frames and returns the last
+        layer features.
+
+    For the example, see the related `documentation <chemiscope-explore-torchscript>`_.
+
+    .. _metatrain: https://lab-cosmo.github.io/metatrain/latest/index.html
+    .. _chemiscope-explore-torchscript:
+        https://chemiscope.org/docs/examples/7-explore-advanced.html#example-with-metatrain-model
+    .. _torch.jit.script:
+        https://pytorch.org/docs/stable/generated/torch.jit.script.html#torch.jit.script
+    """
+    # Check if dependencies were installed
+    try:
+        import torch
+        from metatensor.torch.atomistic import (
+            MetatensorAtomisticModel,
+            ModelEvaluationOptions,
+            ModelOutput,
+            load_atomistic_model,
+            systems_to_torch,
+        )
+        from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+        from metatrain.utils.neighbor_lists import get_system_with_neighbor_lists
+    except ImportError as e:
+        raise ImportError(
+            f"Required package not found: {e}. Please install the dependency using "
+            "'pip install chemiscope[torchscript_featurizer]'."
+        )
+
+    def featurize(frames):
+        """
+        Featurize the given frames using the loaded model.
+
+        :param list frames: A list of atomic structures to obtain systems
+
+        :returns numpy.ndarray: Last layer features for each atom.
+        """
+        opts = {"mtt::aux::last_layer_features": ModelOutput(per_atom=True)}
```

Comment:
I don't like that we are using an output internal to metatrain here. I would prefer to properly define a standard output for "features" and then use that. I'll be in the office tomorrow, we can have a chat then.

I think we could unify this and the output used in PLUMED for machine learning collective variables: https://www.plumed.org/doc-master/user-doc/html/_m_e_t_a_t_e_n_s_o_r.html, look at the "Collective variables and metatensor models" section of the code.